File: backend-laravel/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class Product extends Model
{
    use HasUuids;

    protected $primaryKey = 'id';
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'business_id', 'name', 'sku', 'barcode', 'brand', 'brand_id', 'unit_type',
        'category_id', 'supplier_id', 'buying_price', 'selling_price',
        'stock_quantity', 'low_stock_level', 'reorder_point',
        'expiry_date', 'image_url', 'notes', 'is_active',
    ];

    protected $casts = [
        'buying_price' => 'decimal:2',
        'selling_price' => 'decimal:2',
        'is_active' => 'boolean',
        'expiry_date' => 'date',
        'reorder_point' => 'integer',
    ];

    public function business()
    {
        return $this->belongsTo(Business::class);
    }

    public function sales()
    {
        return $this->hasMany(Sale::class);
    }

    public function category()
    {
        return $this->belongsTo(Category::class)->withDefault();
    }

    public function supplier()
    {
        return $this->belongsTo(Supplier::class)->withDefault();
    }

    public function managedBrand()
    {
        return $this->belongsTo(Brand::class, 'brand_id');
    }

    public function stockMovements()
    {
        return $this->hasMany(StockMovement::class);
    }
}

File: mobile-flutter/lib/core/models/product.dart
class Product {
  final String id;
  final String name;
  final String? sku;
  final double buyingPrice;
  final double sellingPrice;
  final int stockQuantity;
  final int lowStockLevel;
  final bool isActive;

  Product({
    required this.id,
    required this.name,
    this.sku,
    required this.buyingPrice,
    required this.sellingPrice,
    required this.stockQuantity,
    required this.lowStockLevel,
    required this.isActive,
  });

  bool get isLowStock => stockQuantity <= lowStockLevel;

  factory Product.fromJson(Map<String, dynamic> json) {
    final stock = json['stock'] ?? json['stock_quantity'] ?? json['stockQuantity'] ?? 0;
    final lowStock = json['low_stock_level'] ?? json['lowStockLevel'] ?? 5;
    return Product(
      id: json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      sku: json['sku'],
      buyingPrice: _toDouble(json['buying_price'] ?? json['buyingPrice']),
      sellingPrice: _toDouble(json['selling_price'] ?? json['sellingPrice']),
      stockQuantity: stock is int ? stock : int.tryParse(stock.toString()) ?? 0,
      lowStockLevel:
          lowStock is int ? lowStock : int.tryParse(lowStock.toString()) ?? 5,
      isActive: json['is_active'] ?? json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'sku': sku,
      'buying_price': buyingPrice,
      'selling_price': sellingPrice,
      'stock_quantity': stockQuantity,
      'low_stock_level': lowStockLevel,
      'is_active': isActive,
    };
  }

  static double _toDouble(dynamic v) {
    if (v == null) return 0.0;
    if (v is double) return v;
    if (v is int) return v.toDouble();
    return double.tryParse(v.toString()) ?? 0.0;
  }
}
